### PR TITLE
implemented mail queue to send usage notification

### PR DESCRIPTION
**Pull request by varsha766**
## Title
Provide a short, descriptive title of the change (50 characters max recommended).
- Implemented email notification module
- Sending credit expiry and credit exhaust notification to Super Admin
---

## Summary
One-paragraph summary describing what the change does and why. Link related issue(s) or design docs.

---

## Background / Motivation
Explain the context, problem being solved, and any alternatives considered. Include links to design docs, issues, or RFCs.

---

## Scope of Changes
- Files / modules affected (high level)
- Public API changes (if any)
- Database / schema / migration impact (if any)

---

## Type
- [ ] Feature
- [ ] Bugfix
- [ ] Docs
- [ ] Refactor
- [ ] Tests
- [ ] Performance
- [ ] Build / CI / Config

---

## Implementation
Describe the approach taken, key design decisions, and why.

---

## Testing & QA
- Units / integration tests added: describe coverage and important test cases
- Manual / exploratory test steps for QA
- CI matrix: platforms/versions validated

---

## Migration / Rollout Plan
- Data migrations and downtime expectations
- Backwards compatibility notes
- Feature flags and staged rollout instructions

---

## Security / Privacy Considerations
List any security implications, new secrets, or compliance-related items.

---

## Performance Impact
Any benchmark results or expected performance regressions/improvements.

---

## Release Notes (for changelog)
One-line summary suitable for release notes / CHANGELOG.

---

## Checklist (required)
- [ ] Linked issue or ticket present (link: )
- [ ] Target branch is correct
- [ ] PR description explains the why, not just the what
- [ ] Code compiles and tests pass locally
- [ ] Unit and integration tests added where applicable
- [ ] Documentation updated (README, docs site, comments)
- [ ] CHANGELOG.md (or repo-specific changelog) updated with an entry
- [ ] `package.json` / package version updated if publishing a release
- [ ] No secrets or credentials committed
- [ ] Security review completed if required
- [ ] Performance benchmarks included if applicable
- [ ] Migration steps included if applicable

---

## Reviewer Guidance
List the specific areas you want reviewers to focus on (e.g., security, migration, public API, edge cases). Provide commands or steps to run the project locally to validate the change:

```bash
# checkout PR branch
git checkout $BRANCH
# run tests
npm ci && npm test
# run lint
npm run lint
```

---

## Acceptance Criteria
Define the minimal conditions that must be true for this PR to be merged (e.g., tests pass, performance threshold met, migration completed).

---

## Approvals
- [ ] At least one approving review from code owners
- [ ] CI status checks are green
- [ ] Any required manual QA sign-off

### CODE REVIEW

File: src/credit-manager/services/credit-manager.service.ts [hunk 1]
```diff
@@ -225,4 +233,186 @@ export class CreditService {
...
+  async checkAndTriggerNotifications(planId: string) {
+    const plan = await this.creditRepository.findParticularCreditDetail({
+      _id: planId,
+    });
+
+    if (!plan) {
+      return;
+    }
+
+    await this.checkAndTriggerUsageNotification(plan);
```
Comment (Vishwas1) [inline src/credit-manager/services/credit-manager.service.ts]: do we need to make them await ?

File: src/credit-manager/services/credit-manager.service.ts [hunk 2]
```diff
@@ -225,4 +233,186 @@ export class CreditService {
...
+    try {
+      if (!plan.expiresAt) return;
+
+      const remainingDays = Math.max(
+        Math.ceil(
+          (plan.expiresAt.getTime() - Date.now()) / (1000 * 60 * 60 * 24),
+        ),
+        0,
+      );
+      const thresholds = this.configService
```
Comment (Vishwas1) [inline src/credit-manager/services/credit-manager.service.ts]: Can we prepare this outside the fucntion so that we dont need to do calculation every time we send email

File: src/credit-manager/services/credit-manager.service.ts [hunk 3]
```diff
@@ -225,4 +233,186 @@ export class CreditService {
...
+    Logger.log(
+      'Inside checkAndTriggerUsageNotification() to send notification',
+      'CreditManagerService',
+    );
+    try {
+      const { totalCredits, used, serviceId } = plan;
+      if (!totalCredits || totalCredits <= 0) return;
+      const usedCredits = used || 0;
+      const usedPercentage = Math.floor((usedCredits / totalCredits) * 100);
+      const thresholds = this.configService
```
Comment (Vishwas1) [inline src/credit-manager/services/credit-manager.service.ts]: Can we prepare this outside the fucntion so that we dont need to do calculation every time we send email

File: src/credit-manager/services/credit-manager.service.ts [hunk 4]
```diff
@@ -225,4 +233,186 @@ export class CreditService {
...
+      );
+      if (!updatedResult) {
+        Logger.log(
+          `Skipping credit usage notification | serviceId=${serviceId} | threshold=${thresholdToNotify} | Reason: Already notified or no active plan found`,
+        );
+        return;
+      }
+
+      // 🚀 Push job to BullMQ
+      const notificationQueue =
```
Comment (Vishwas1) [inline src/credit-manager/services/credit-manager.service.ts]: this part should before updating the db entry

File: src/credit-manager/services/credit-manager.service.ts [hunk 5]
```diff
@@ -225,4 +233,186 @@ export class CreditService {
...
+          `Skipping credit expiry notification | serviceId=${plan.serviceId} | threshold=${thresholdToNotify} | Reason: Already notified or no active plan found`,
+        );
+        return;
+      }
+
+      const notificationQueue =
+        this.configService.get('DASHBOARD_NOTIFICATION_QUEUE') ||
+        'Credit-Notification-Queue';
+
+      await this.mailClientService.addAJob(
```
Comment (Vishwas1) [inline src/credit-manager/services/credit-manager.service.ts]: this part should before updating the db entry

File: src/mailClient/controller/mail-client.controller.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Controller, Post, Body } from '@nestjs/common';
+import { MailClientService } from '../service/mail-client.service';
+
+@Controller('mail-client')
+export class MailClientController {
+  constructor(private readonly bullQueueService: MailClientService) {}
+
+  @Post('add-job')
```
Comment (Vishwas1) [inline src/mailClient/controller/mail-client.controller.ts:14]: This api should not be exposed to public ..please confirm

File: env.sample [hunk 7]
```diff
@@ -34,3 +34,10 @@ SERVICE_SUFFIX="SSI_API"
 NODE_ENV= 'production'
 // development
 
+//notification
+
+
```
Comment (Vishwas1) [inline env.sample:42]: added descirptions for each of these vars in comment

**Review by Vishwas1 — APPROVED** [review]

File: src/mailClient/controller/mail-client.controller.ts (hunk 6 above)
Comment (varsha766) [reply src/mailClient/controller/mail-client.controller.ts:14]: This API is not publicly accessible.
